### PR TITLE
Achieve 100% test coverage

### LINKED ISSUE #369: Achieve 100% test coverage
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

I'd like to try to increase test coverage, are you interested?

### PR DESCRIPTION
**Pull request by jean-michelet**
Fixed #369

100%-test-coverage-fastify-autoload

I failed to cover some branches, so I have removed them. If you can find a way to fully evaluate these branches, please let me know. However, it's possible that they may be obsolete.

I've created an `issues` folder for issues related tests, which will need to follow this pattern: `test/issues/<issue-number>/test.js`. Since we're testing the autoloader, we need to create fixtures. This is why I chose to create a folder named after the issue instead of placing `<issue-number>.js` directly.

If you agree with this approach, I will update [this PR](https://github.com/fastify/fastify-autoload/pull/368) to reflect these changes.

### CODE REVIEW

**Comment by jean-michelet** [comment]
Um...

I'm going to investigate.

**Comment by jean-michelet** [comment]
Waiting for #371  to be merged.

File: index.js [hunk 1]
```diff
@@ -6,7 +6,9 @@ const { pathToFileURL } = require('node:url')
 
 const isFastifyAutoloadTypescriptOverride = !!process.env.FASTIFY_AUTOLOAD_TYPESCRIPT
 const isTsNode = (Symbol.for('ts-node.register.instance') in process) || !!process.env.TS_NODE_DEV
-const isBabelNode = (process?.execArgv || []).concat(process?.argv || []).some((arg) => arg.indexOf('babel-node') >= 0)
+const isBabelNode = (/* istanbul ignore next */process.execArgv || [])
```
[the remark this answers is not in the document]
Comment (gurgunday) [reply index.js:9]: Don't think they are necessary

**Comment by Uzlopak** [comment]
You are targetting master, but #371 is targetting next

**Comment by mcollina** [comment]
Can we achieve this on master instead? I think it would be a moving target on next.

**Comment by jean-michelet** [comment]
> Can we achieve this on master instead? I think it would be a moving target on next.

Should I try to fix the [failing TS test](https://github.com/fastify/fastify-autoload/actions/runs/8754436532/job/24027590368) here?
Or it should be fixed #371?

**Comment by gurgunday** [comment]
The test that's failing here and the one in 371 are different I think

So for this PR, I'd suggest making the tests pass and merging to `master`, and then focusing on `next`'s failing test at 371

**Comment by jean-michelet** [comment]
> The test that's failing here and the one in 371 are different I think

Yep, I realize got completely lost since your comment, my bad...

**Comment by jean-michelet** [comment]
I pushed but I noticed that [L17](https://github.com/fastify/fastify-autoload/pull/370/files#diff-e727e4bdf3657fd1d798edcd6b099d6e092f8573cba266154583a746bba0f346L17) not covered when tests running on windows...

I will need to investigate...

File: index.js [hunk 2]
```diff
@@ -6,16 +6,17 @@ const { pathToFileURL } = require('node:url')
...
+const isBabelNode = process.execArgv.concat(process.argv).some((arg) => arg.indexOf('babel-node') >= 0)
 
 const isVitestEnvironment = process.env.VITEST === 'true' || process.env.VITEST_WORKER_ID !== undefined
 const isJestEnvironment = process.env.JEST_WORKER_ID !== undefined
-const isSWCRegister = process._preload_modules && process._preload_modules.includes('@swc/register')
-const isSWCNodeRegister = process._preload_modules && process._preload_modules.includes('@swc-node/register')
+const isSWCRegister = process._preload_modules?.includes('@swc/register')
+const isSWCNodeRegister = process._preload_modules?.includes('@swc-node/register')
+/* istanbul ignore next - OS specific */
 const isSWCNode = typeof process.env._ === 'string' && process.env._.includes('.bin/swc-node')
```
Comment (jean-michelet) [inline index.js:16]: Never evaluated on windows...

**Comment by jean-michelet** [comment]
Ready For Review

**Review by mcollina — APPROVED** [review]
lgtm

**Review by gurgunday — APPROVED** [review]
lgtm

**Pull request merged**